dwave/cloud/computation.py

```diff
@@ -153,6 +153,27 @@ def wait_multiple(futures, min_done=None, timeout=None):
 
         Returns:
             2-tuple of futures done and not_done, similar to `concurrent.futures.wait()`
+
+        Examples:
+
+        By default all sampling requests are processed asynchronously. Reading results from
+        any future object is a blocking operation.
+
+        .. code-block:: python
+
+            # We can submit several sample requests without blocking
+            # (In this specific case we could accomplish the same thing by increasing 'num_reads')
+            futures = [solver.sample_ising(linear, quad, num_reads=100) for _ in range(10)]
+
+            # We can check if a set of samples are ready without blocking
+            print(futures[0].done())
+
+            # We can wait on a single future
+            futures[0].wait()
+
+            # Or we can wait on several futures
+            dwave.cloud.computation.Future.wait_multiple(futures, min_done=3)
+            
```

Comment:
Heads up: mention accessing `future.samples` or `future['samples']` is a blocking operation for the first time (calls `.result()`). Next access to any of the properties is non-blocking.

Also, `as_completed` is modeled after `concurrent.futures.as_completed` (has same semantics as much as possible). `as_completed` is usually preferred over `wait_multiple` since it is a blocking iterable of resolved futures.